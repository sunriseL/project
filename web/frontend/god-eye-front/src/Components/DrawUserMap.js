import React from 'react';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import '../App.css';
import $ from "jquery";
import { Grid, Input, InputLabel } from '../../node_modules/@material-ui/core';
import './UserMap.css';

const styles = {
    card: {
        maxWidth: 2160,
        width: '60%'
    },
    media: {
        height: '60%',
        paddingTop: '56.25%', // 16:9
    },
};

function select(){
    document.getElementById('map-path').value = document.getElementById('image').value;
}

function huayuan(){
    var c=document.getElementById("canvas");
    var ctx = c.getContext('2d');
    ctx.beginPath();
    ctx.arc(150,75,10,0,2*Math.PI);
    ctx.fillStyle="red";
    ctx.fill();
}

class DrawUserMap extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            map_name: "用户地图",
            map_bin: localStorage.getItem('currentMapBin'),
        };
    };

    componentDidMount(){
        let canvas = document.getElementById('canvas');
        canvas.width = 1000;
        canvas.height= 800;
        let ctx = canvas.getContext('2d');
        let img = new Image();
        img.src = this.state['map_bin'];
        img.onload=function(){ctx.drawImage(img,0,0,1000,800)};
    }

    getBase64(file,cb){
        if(typeof(FileReader) === 'undefined'){
            alert("您的浏览器不支持FileReader,请使用Chrome访问本应用");
            return;
        }

        let reader = new FileReader();
        reader.readAsDataURL(file);

        reader.onload = function(){
            cb(reader.result);
        }

        reader.onerror = function(error){
            console.log('Error: ', error);
        }
    }

    upload() {
        let file = document.getElementById('image').files[0];

        this.getBase64(file, (result) => {
            let map_name = document.getElementById('mapNameHolder').value;
            let uploadJSON = {
                'map_bin': result,
                'map_name': map_name,
            };
            this.setState(uploadJSON);
            localStorage.setItem('currentMapBin',result);
            let _this=this;
            $.ajax({
                type: "post",
                url: "http://127.0.0.1:8081/map/add",
                crossDomain: true,
                data: uploadJSON,
                success: function (data) {
                    alert("地图上传成功");
                    _this.render();
                },
                error : function() {
                    alert("地图名重复，请使用其它地图名");
                }
            })
        });
        this.render();
    }

    ifSetting(){
        var url = document.location.toString();
        var arrUrl = url.split("//");
        var splitUrl = arrUrl[1].split("/");
        var relUrl = splitUrl[1];//stop省略，截取从start开始到结尾的所有字符

        if(relUrl.indexOf("?") !== -1){
            relUrl = relUrl.split("?")[0];
        }
        return (relUrl==='settings');

    }

    render(){
        let mapInstantce = <div>Loading</div>

        if(!(localStorage.getItem('ifDBEmpty')==='true')){
            mapInstantce = <canvas id="canvas" ></canvas>
        }

        return (
            <div>
                <Card style={{margin: "1%"}}>

                    <Button id="snap" onClick={() => huayuan()}>画图</Button>
                    {mapInstantce}
                    <CardContent>
                        <Typography gutterBottom variant="headline" component="h2">
                            { this.state['map_name'] }
                        </Typography>
                    </CardContent>
                </Card>
                {(this.ifSetting() &&
                    <Grid container>
                        <Grid item xs={2} />
                        <Grid item xs={2} style={{position: 'relative'}}>
                            <TextField
                                id="mapNameHolder" class="NameHolder"
                                label="请输入地图名称"
                                placeholder="请输入地图名称"
                                margin="normal"
                                style={{position: 'absolute', top: 0, left: 0}}
                            />
                        </Grid>
                        <Grid id="control-grid" item xs={2} className="control-grid">
                            <input type="file"  id="image" onChange={() => select()} />
                            <Button variant="contained" small >选择文件</Button>
                        </Grid>
                        <Grid item xs={3}>
                            <InputLabel>文件名</InputLabel>
                            <Input id="map-path" disabled/>
                        </Grid>
                        <Grid item xs={2}>
                            <Button variant="contained" small onClick = {() => this.upload()}>上传地图</Button>
                        </Grid>
                    </Grid>)}
            </div>
        );
    }
}

export default withStyles(styles)(DrawUserMap);