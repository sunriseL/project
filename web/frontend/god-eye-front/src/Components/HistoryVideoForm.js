import React from 'react';
import { Form, DatePicker,  Button, Row, Col } from 'antd';

const FormItem = Form.Item;
const RangePicker = DatePicker.RangePicker;

class HistoryVideoForm extends React.Component {
    handleSubmit = (e) => {
        e.preventDefault();

        this.props.form.validateFields((err, fieldsValue) => {
        if (err) {
            return;
        }

        // Should format date value before submit.
        const rangeTimeValue = fieldsValue['range-time-picker'];
        const values = {
            ...fieldsValue,
            'range-time-picker': [
            rangeTimeValue[0].format('YYYY-MM-DD HH:mm:ss'),
            rangeTimeValue[1].format('YYYY-MM-DD HH:mm:ss'),
            ],
        };

        console.log('Received values of form: ', values);
        // Commit time data to server here
        });
    }

    render() {
        const { getFieldDecorator } = this.props.form;

        const rangeConfig = {
        rules: [{ type: 'array', required: true, message: 'Please select time!' }],
        };
        return (
        <Form onSubmit={this.handleSubmit}>
            <Row>
                <Col span={6} offset={6}>
                    <FormItem>
                        {getFieldDecorator('range-time-picker', rangeConfig)(
                            <RangePicker showTime format="YYYY-MM-DD HH:mm:ss" />
                        )}
                    </FormItem>
                </Col>
                <Col span={6}>
                    <FormItem>
                        <Button type="primary" htmlType="submit">Submit</Button>
                    </FormItem>
                </Col>
            </Row>
        </Form>
        );
    }
}

const WrappedHistoryVideoForm = Form.create()(HistoryVideoForm);
export default WrappedHistoryVideoForm;