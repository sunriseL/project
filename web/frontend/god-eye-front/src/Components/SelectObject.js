import React from 'react';
import '../App.css';
import { Grid } from '../../node_modules/@material-ui/core';

class SelectObject extends React.Component {
    render(){
        return(
            <Grid  container xs={12} spacing={16}>
                <Grid item xs  alignItems={"center"}>
                    <canvas id="screenShot" hidden/>
                </Grid>
                <Grid item xs spacing={16}>
                    <canvas id = "selectedPart" hidden/>
                </Grid>
            </Grid>
        );
    }
}

export default SelectObject;