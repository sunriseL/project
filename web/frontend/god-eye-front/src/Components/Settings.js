import React from 'react';
import {  Modal,Layout, Upload, Icon, message, Breadcrumb } from 'antd';
import UserMap from './UserMap';
import MyUpload from "./Upload";

const uploadProps = {
    name: 'file',
    multiple: false,
    action: 'localhost:8090',
    onChange(info) {
        const status = info.file.status;
        if (status !== 'uploading') {
            console.log(info.file, info.fileList);
        }       
        if (status === 'done') {
            message.success(`${info.file.name} file uploaded successfully.`);
        } else if (status === 'error') {
            message.error(`${info.file.name} file upload failed.`);
        }
    },
}

const {  Content } = Layout;
const Dragger = Upload.Dragger;

function getBase64(img, callback) {
    const reader = new FileReader();
    reader.addEventListener('load', () => callback(reader.result));
    reader.readAsDataURL(img);
}

function beforeUpload(file) {
    const isJPG = file.type === 'image/jpeg';
    if (!isJPG) {
        message.error('You can only upload JPG file!');
    }
    const isLt2M = file.size / 1024 / 1024 < 2;
    if (!isLt2M) {
        message.error('Image must smaller than 2MB!');
    }
    return isJPG && isLt2M;
}

class Settings extends React.Component {
    constructor() {
        super();
        this.state = {
            loading: false,
            previewVisible: false,
            previewImage: '',
            fileList: [{
                uid: -1,
                name: 'xxx.png',
                status: 'done',
                url: 'https://ss1.bdstatic.com/70cFvXSh_Q1YnxGkpoWK1HF6hhy/it/u=930073930,3424315015&fm=27&gp=0.jpg',
            }],
        }
    }

    handleChange1 = (info) => {
        if (info.file.status === 'uploading') {
            this.setState({ loading: true });
            return;
        }
        if (info.file.status === 'done') {
            // Get this url from response in real world.
            getBase64(info.file.originFileObj, imageUrl => this.setState({
                imageUrl,
                loading: false,
            }));
        }
    }

    render(){
        const uploadButton = (
            <div>
                <Icon type={this.state.loading ? 'loading' : 'plus'} />
                <div className="ant-upload-text">Upload</div>
            </div>
        );
        const imageUrl = this.state.imageUrl;
        return(
            <Layout>
                <Layout>
                    <Breadcrumb>
                        <Breadcrumb.Item>慧眼识踪</Breadcrumb.Item>
                        <Breadcrumb.Item>导入地图与摄像头</Breadcrumb.Item>
                    </Breadcrumb>
                </Layout>
                <Content>
                    <UserMap />
                    <Dragger {...uploadProps}>
                        <p className="ant-upload-drag-icon">
                            <Icon type="inbox" />
                        </p>
                        <p className="ant-upload-drag-text">上传地图</p>
                        <p className="ant-upload-drag-hint">点击或拖拽上传地图图片</p>
                    </Dragger>
                    <MyUpload/>
                    <Upload
                        name="avatar"
                        listType="picture"
                        className="avatar-uploader"
                        showUploadList={false}
                        action="//jsonplaceholder.typicode.com/posts/"
                        beforeUpload={beforeUpload}
                        onChange={this.handleChange1}
                    >
                        {imageUrl ? <img src={imageUrl} alt="avatar" /> : uploadButton}
                    </Upload>
                </Content>
            </Layout>
        );
    }
}

export default Settings;