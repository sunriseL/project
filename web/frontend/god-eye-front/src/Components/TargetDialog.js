import React from 'react';
//import { Dialog, Grid, Typography } from '@material-ui/core/es';
import Dialog from '@material-ui/core/Dialog';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import $ from 'jquery';
import { DialogContent } from '@material-ui/core';

class TargetDialog extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            targetList: [],
        }
    }
    
    getTargetList(){
        $.ajax({
            type: "get",
            url: "http://127.0.0.1:8081/target/get",
            crossDomain: true,
            async:true,
            success: function (data) {
                this.setState({targetList: data});
            },
            error : function(error) {
                console.log(error);
            }
        });
    }

    componentWillMount(){
        this.getTargetList();
    }

    handleSelect = (imgSrc) =>{
        this.props.onClose();
        let picJson = {
            imgStream: imgSrc,
        };
        $.ajax({
            type: 'post',
            url: 'http://127.0.0.1:8081/target/choose',
            crossDomain: true,
            async: true,
            data: picJson,
            success: function(data) {
                this.drawRoute(data);
            },
            error: function (error){
                console.log(error);               
            },
        })
    }

    handleClose = () =>{
        this.props.onClose();
    }

    drawRoute(data){
        // data should be formatted as JSONArray as [{cameraId: , x: , y: , relativeTime: , absoluteTime , }]
        let ctx = document.getElementById('lightCameraCanvas');
        let ctxL = ctx.height;
        let ctxW = ctx.width;
        for(let i=0; i<data.length; i++){
            ctx.arc(i.x * ctxW, i.y * ctxL, 1, 0, 2 * Math.PI);
            ctx();
        }
    }


    render(){
        const { onClose, ...other } = this.props;

        let items = null;
        if(this.state.targetList.length === 0){
            items = <Typography variant='headline' style={{margin: "5%"}}>尚未处理完成，请耐心等待</Typography>;
        }else{
            items = <Grid container spacing={8}>{this.state.targetList.map(i => (
                <Grid item xs onClick={this.handleSelect(i.imgsrc)}><img src={i.imgsrc} alt='无法显示图片' /></Grid>
            ))}</Grid>
        }

        return(
            <Dialog onClose={this.handleClose} {...other}>
                <DialogContent>
                    {items}
                </DialogContent>
            </Dialog>
        )
    }
}

export default TargetDialog;