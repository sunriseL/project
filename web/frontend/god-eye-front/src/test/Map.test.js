import React from 'react'
import { shallow ,render} from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import HistoryVideoForm from "../Components/HistoryVideoForm"
import TargetDialog from "../Components/TargetDialog";
import DrawUserMap from "../Components/DrawUserMap";
import SelectObject from "../Components/SelectObject";
import Settings from "../Components/Settings";
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

const wrapper = shallow(<HistoryVideoForm />);
const wrapper1 = shallow(<TargetDialog />);

describe('HistoryVideoForm', () => {
    it('HistoryVideoForm Component should be render', () => {
        expect(wrapper.find('FormItem').exists());
    });
});

describe('TargetDialog', () => {
    it('TargetDialog Component should be render', () => {
        expect(wrapper1.find('Dialog').exists());
    });
});

describe('DrawUserMap', () => {
    it('DrawUserMap Component should be render', () => {
        expect(shallow(<DrawUserMap />).find('Grid').exists());
    });
});

describe('<SelectObject/>', () => {
    it('SelectObject Component should be render', () => {
        const tree = renderer
            .create("SelectObject").toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Settings', () => {
    it('DrawUserMap Component should be render', () => {
        expect(shallow(<Settings />).find('BreadCrumb').exists());
    });
});

