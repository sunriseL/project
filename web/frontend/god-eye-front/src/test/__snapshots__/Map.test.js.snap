// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectObject/> SelectObject Component should be render 1`] = `"SelectObject"`;
